components/system/components/Typography.js

```diff
@@ -18,6 +18,31 @@ const LINK_STYLES = `
   }
 `;
 
+const useColorProp = (color) =>
+  React.useMemo(
+    () => (theme) => {
+      if (!color) return;
+      if (!(color in theme.system)) {
+        console.warn(`${color} doesn't exist in our design system`);
+        return;
+      }
+      return css({ color: theme.system[color] });
+    },
+    [color]
+  );
+
+const truncateElements = (nbrOfLines) =>
+  nbrOfLines &&
+  css`
+    overflow: hidden;
+    line-height: 1.5;
+    word-break: break-all;
```

Comment:
I'd recommend using word-break: break-all and overflow-wrap: break-word instead. break-all breaks things off in the middle of a word which looks strange